File: 백준/Silver/2217. 로프/로프.py
import sys

N = int(sys.stdin.readline().strip())
arr = []
for i in range(N):
  arr.append(int(sys.stdin.readline().strip()))
  
arr.sort(reverse=True)
result = []
for i in range(N):
  result.append(arr[i] * (i+1))

print(max(result))

File: 백준/Silver/2217. 로프/로프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(bf.readLine());
        int[] rope = new int[N];

        for (int i = 0; i < N; i++) {
            rope[i] = Integer.parseInt(bf.readLine());
        }
        Arrays.sort(rope);

        int max = Integer.MIN_VALUE;

        for (int i = 0; i < N; i++) {
            max = Math.max(max, rope[i] * (N - i));
        }

        System.out.println(max);

    }
}